### PR TITLE
ci: Run tests with rust nightly and beta channels on a schedule

### PR DESCRIPTION
**Pull request by flub**
This runs only the unittests once a day using the nightly and the beta
channel.  This should not block PRs but will give us earlier warnings
when we need to adjust things.

No caching since the compiler will be different every day anyway and
it is not blocking PRs anyway.

#skip-changelog

### CODE REVIEW

**Review by Swatinem — APPROVED** [review]
can we run clippy as well? since they introduce new breaking lints quite frequently

**Comment by flub** [comment]
Yeah, i wondered if this would be too noisy.  But i'm also just fixing clippy stuff in symbolicator... and since we run normal ci on latest stable this blocks new PRs as soon as a new rust is released. so let's add it and try out i guess

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Rust nightly CI
```
Comment (jan-auer) [inline .github/workflows/nightly.yml:1]: Aligning with other task names. This is evidently Rust.

```suggestion
name: Nightly CI
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - cron: '11 7 * * *'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [nightly, beta]
+    env:
+      RUSTFLAGS: -Dwarnings
```
Comment (jan-auer) [inline .github/workflows/nightly.yml:14]: Please put this on top-level, like in `ci.yml`.

**Review by jan-auer — CHANGES_REQUESTED** [review]
I would be ok with running clippy since usually they only make lints required for features that have been released for 1-2 versions. I might be mistaken though, in which case clippy would just be throwing us off.

**Comment by flub** [comment]
CI failures here are fixed by #297 btw, not because this PR is bad.

**Review by jan-auer — APPROVED** [review]
Should be good after merging master.

**Pull request merged**